### ISSUE TITLE
kontena master login without a responsive master should give better error

### ISSUE THREAD

**Issue by matti** [kontena/kontena#1219]
currently the only error is:

```
[error] Code exchange failed
```

should say "could not connect to master"

**Comment by colinmollenhour** [comment]
I ran into some similar issues..

```
$ kontena master login --code ***REDACTED*** https://***.***.***.***/
 [error] Code exchange failed
$ DEBUG=true kontena master login --code ***REDACTED*** https://***.***.***.***/                
Running #<Kontena::MainCommand:0x00000002689eb0> -- callback matcher = 'nil'
Running #<Kontena::Cli::MasterCommand:0x0000000267dd18> -- callback matcher = 'nil'
Running #<Kontena::Cli::Master::LoginCommand:0x00000002676900> -- callback matcher = 'master login'
D, [2016-10-26T18:56:23.204639 #2401] DEBUG -- CONFIG: Loading configuration from /home/colin/.kontena_client.json
D, [2016-10-26T18:56:23.205085 #2401] DEBUG -- CONFIG: Configuration loaded with 1 servers.
D, [2016-10-26T18:56:23.205159 #2401] DEBUG -- CONFIG: Current master: (not selected)
D, [2016-10-26T18:56:23.205202 #2401] DEBUG -- CONFIG: Current grid: (not selected)
/home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/lib/kontena/cli/master/login_command.rb:135:in `block in use_authorization_code': Code exchange failed (Kontena::Errors::StandardError)
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/lib/kontena/cli/common.rb:80:in `vspinner'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/lib/kontena/cli/master/login_command.rb:118:in `use_authorization_code'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/lib/kontena/cli/master/login_command.rb:33:in `execute'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/lib/kontena/command.rb:181:in `run'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/clamp-1.1.1/lib/clamp/subcommand/execution.rb:11:in `execute'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/lib/kontena/command.rb:181:in `run'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/clamp-1.1.1/lib/clamp/subcommand/execution.rb:11:in `execute'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/lib/kontena/command.rb:181:in `run'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/clamp-1.1.1/lib/clamp/command.rb:132:in `run'
        from /home/colin/.rvm/gems/ruby-2.3.0/gems/kontena-cli-0.16.0/bin/kontena:16:in `<top (required)>'
        from /home/colin/.rvm/gems/ruby-2.3.0/bin/kontena:23:in `load'
        from /home/colin/.rvm/gems/ruby-2.3.0/bin/kontena:23:in `<main>'
        from /home/colin/.rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `eval'
        from /home/colin/.rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `<main>'
$ kontena master login --code ***REDACTED*** https://***.***.***.***/                           
 [error] Code exchange failed
$ curl -k https://***.***.***.***/                                                            
{"name":"Kontena Master","tagline":"The Container Platform","version":"0.16.0"}
$ export SSL_IGNORE_ERRORS=true
$ kontena master login --code ***REDACTED*** https://***.***.***.***/                           
Current master not selected
$ k master ls
Name                     Url                           
 [error] undefined method `name' for nil:NilClass
         Rerun the command with environment DEBUG=true set to get the full exception
$ vi ~/.kontena_client.json 
# The {{.servers[0].name}} was null, I changed it to "testy"
$ k master ls                                                                               
Name                     Url                           
* testy                  https://***.***.***.***/        
```

So the CLI was broken until I manually added the "name" to the ~/.kontena_client.json

**Comment by jakolehm** [comment]
Fixed in #1222
